.github/workflows/deploy-cluster.yml

```diff
@@ -113,3 +113,18 @@ jobs:
       shell: bash
       env:
         TF_VAR_azure_sp_credentials_json: ${{ secrets.AZURE_CREDENTIALS }}
+
+  notify-slack-on-failure:
+    name: Notify Slack on Failure
+    runs-on: ubuntu-latest
+    if: failure()
```

Comment:
This only works if a step failed in the same job: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#failure